### PR TITLE
Emir/host arithmetic

### PR DESCRIPTION
**Pull request by emirsoyturk**

### CODE REVIEW

**Review by nonam3e — APPROVED** [review]

File: wrappers/golang/internal/generator/fields/templates/field_test.go.tmpl [hunk 1]
```diff
@@ -100,6 +100,36 @@ func test{{capitalize .Field}}GenerateScalars(suite *suite.Suite) {
...
+    scalarA := scalarsA[0]
+    square := scalarA.Sqr()
+    mul := scalarA.Mul(&scalarA)
+
+    suite.Equal(square, mul, "Square and multiplication do not yield the same value")
+
+    inv := scalarA.Inv()
+
+    one := scalarA.Mul(&inv)
+	expectedOne := scalarsA[1]
```
Comment (jeremyfelder) [inline wrappers/golang/internal/generator/fields/templates/field_test.go.tmpl:126]: Its cleaner to use a new scalar variable even though it adds a new variable.

File: wrappers/golang/internal/generator/fields/templates/field_test.go.tmpl [hunk 2]
```diff
@@ -100,6 +100,36 @@ func test{{capitalize .Field}}GenerateScalars(suite *suite.Suite) {
...
+	expectedOne.One()
+
+    suite.Equal(one, expectedOne)
```
Comment (jeremyfelder) [inline wrappers/golang/internal/generator/fields/templates/field_test.go.tmpl:129]: The first argument is meant be the "expected" value

File: wrappers/rust/icicle-core/src/traits.rs [hunk 3]
```diff
@@ -30,3 +31,8 @@ pub trait MontgomeryConvertible: Sized {
     fn to_mont(values: &mut (impl HostOrDeviceSlice<Self> + ?Sized), stream: &IcicleStream) -> eIcicleError;
     fn from_mont(values: &mut (impl HostOrDeviceSlice<Self> + ?Sized), stream: &IcicleStream) -> eIcicleError;
 }
+
+pub trait Arithmetic: Sized + Add<Output = Self> + Sub<Output = Self> + Mul<Output = Self> {
```
Comment (yshekel) [inline wrappers/rust/icicle-core/src/traits.rs:35]: why not put it in fieldImpl trait?

**Review by jeremyfelder — APPROVED** [review]

**Review by LeonHibnik — APPROVED** [review]

File: wrappers/rust/icicle-core/src/field.rs [hunk 4]
```diff
@@ -106,6 +107,61 @@ pub trait MontgomeryConvertibleField<F: FieldImpl> {
...
+    fn sqr(self) -> Self {
+        F::sqr(self)
+    }
+
+    fn inv(self) -> Self {
+        F::inv(self)
+    }
+}
+
+impl<const NUM_LIMBS: usize, F: FieldConfig> Add for Field<NUM_LIMBS, F>
```
[the remark this answers is not in the document]
Comment (LeonHibnik) [reply wrappers/rust/icicle-core/src/field.rs:132]: Passing a reference will require redesigning the Field traits, and possibly introduce some breaking changes TODO for v4

**Pull request merged**